### ISSUE TITLE
IDE0028 needs to be split into two for cases where semantics change and for cases where it does not

### ISSUE THREAD

**Issue by nvmkpk** [dotnet/roslyn#73104]
Version Used:
Visual Studio 17.10.0 Preview 4.0

Steps to Reproduce:

Add code like below:
```
IList<string> strings = new SortedStringCollection();
```

Diagnostic Id: IDE0028

Expected Behavior:
We should be able to globally disable IDE0028 for cases where it changes the semantics while it is not disabled for other cases.

Actual Behavior:
IDE0028 is reported and if suggestion is applied, it changes the code to as shown below:
```
IList<string> strings = new [];
```
Compiler generates below code for:
```
IList<string> strings = new List<string>();
```
If `SortedStringCollection` inserts the items in a specific sorted order, that will no longer happen when IDE0028 is applied.
